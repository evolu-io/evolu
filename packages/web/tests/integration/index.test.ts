import { serializeDom } from './helpers/serializeDom';

beforeEach(async () => {
  await page.goto(`file://${__dirname}/out/index.html`);
  await page.waitFor(50);
});

test('page title', async () => {
  await expect(page.title()).resolves.toMatch('Slad');
});

test('render', async () => {
  await expect(await page.evaluate(serializeDom)).toMatchSnapshot();
});

test('focus blur via tab, tab, tab, click to focus button, click to blur button', async () => {
  await page.keyboard.press('Tab');
  await page.waitFor(50);
  await expect(await page.evaluate(serializeDom)).toMatchSnapshot();

  await page.keyboard.press('Tab');
  await page.waitFor(50);
  await expect(await page.evaluate(serializeDom)).toMatchSnapshot();

  await page.keyboard.press('Tab');
  await page.waitFor(50);
  await expect(await page.evaluate(serializeDom)).toMatchSnapshot();

  await page.click('button.focus');
  await page.waitFor(50);
  await expect(await page.evaluate(serializeDom)).toMatchSnapshot();

  await page.click('button.blur');
  await page.waitFor(50);
  await expect(await page.evaluate(serializeDom)).toMatchSnapshot();
});

// https://github.com/steida/slad/issues/14
export default undefined;
