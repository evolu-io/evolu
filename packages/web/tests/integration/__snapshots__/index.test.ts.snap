// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`focus blur via tab, tab, tab, click to focus button, click to blur button 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-2517315232"
  >
    <div
      class="jsx-3659724228"
    >
      ·Slad·
    </div>
    <div
      class="jsx-442929638"
    >
      ·Basic Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          style="font-size:24px"
        >
          ·heading·
        </div>
        <div
          style="font-size:16px"
        >
          ·paragraph·
        </div>
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          ·{"anchor":[0,0,0],"focus":[0,0,0]}·
          <br />
          ·hasFocus: ·
          ·true·
        </pre>
      </div>
    </div>
    <div
      class="jsx-442929638"
    >
      ·Schema Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          class="jsx-3102962925"
        >
          ·heading·
        </div>
        <div
          class="jsx-1758464876"
        >
          ·paragraph·
        </div>
        <div
          class="jsx-44984361"
        >
          <div
            class="jsx-1758464876"
          >
            ·listitem·
          </div>
        </div>
        <img
          alt="Square placeholder image 80px"
          height="80"
          src="https://via.placeholder.com/80"
          width="80"
        />
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <button
      class="focus"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`focus blur via tab, tab, tab, click to focus button, click to blur button 2`] = `
<div
  id="__next"
>
  <div
    class="jsx-2517315232"
  >
    <div
      class="jsx-3659724228"
    >
      ·Slad·
    </div>
    <div
      class="jsx-442929638"
    >
      ·Basic Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          style="font-size:24px"
        >
          ·heading·
        </div>
        <div
          style="font-size:16px"
        >
          ·paragraph·
        </div>
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <div
      class="jsx-442929638"
    >
      ·Schema Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          class="jsx-3102962925"
        >
          ·heading·
        </div>
        <div
          class="jsx-1758464876"
        >
          ·paragraph·
        </div>
        <div
          class="jsx-44984361"
        >
          <div
            class="jsx-1758464876"
          >
            ·listitem·
          </div>
        </div>
        <img
          alt="Square placeholder image 80px"
          height="80"
          src="https://via.placeholder.com/80"
          width="80"
        />
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          ·{"anchor":[0,0,0],"focus":[0,0,0]}·
          <br />
          ·hasFocus: ·
          ·true·
        </pre>
      </div>
    </div>
    <button
      class="focus"
      disabled=""
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`focus blur via tab, tab, tab, click to focus button, click to blur button 3`] = `
<div
  id="__next"
>
  <div
    class="jsx-2517315232"
  >
    <div
      class="jsx-3659724228"
    >
      ·Slad·
    </div>
    <div
      class="jsx-442929638"
    >
      ·Basic Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          style="font-size:24px"
        >
          ·heading·
        </div>
        <div
          style="font-size:16px"
        >
          ·paragraph·
        </div>
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <div
      class="jsx-442929638"
    >
      ·Schema Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          class="jsx-3102962925"
        >
          ·heading·
        </div>
        <div
          class="jsx-1758464876"
        >
          ·paragraph·
        </div>
        <div
          class="jsx-44984361"
        >
          <div
            class="jsx-1758464876"
          >
            ·listitem·
          </div>
        </div>
        <img
          alt="Square placeholder image 80px"
          height="80"
          src="https://via.placeholder.com/80"
          width="80"
        />
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          ·{"anchor":[0,0,0],"focus":[0,0,0]}·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <button
      class="focus"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`focus blur via tab, tab, tab, click to focus button, click to blur button 4`] = `
<div
  id="__next"
>
  <div
    class="jsx-2517315232"
  >
    <div
      class="jsx-3659724228"
    >
      ·Slad·
    </div>
    <div
      class="jsx-442929638"
    >
      ·Basic Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          style="font-size:24px"
        >
          ·heading·
        </div>
        <div
          style="font-size:16px"
        >
          ·paragraph·
        </div>
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <div
      class="jsx-442929638"
    >
      ·Schema Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          class="jsx-3102962925"
        >
          ·heading·
        </div>
        <div
          class="jsx-1758464876"
        >
          ·paragraph·
        </div>
        <div
          class="jsx-44984361"
        >
          <div
            class="jsx-1758464876"
          >
            ·listitem·
          </div>
        </div>
        <img
          alt="Square placeholder image 80px"
          height="80"
          src="https://via.placeholder.com/80"
          width="80"
        />
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          ·{"anchor":[0,0,0],"focus":[0,0,0]}·
          <br />
          ·hasFocus: ·
          ·true·
        </pre>
      </div>
    </div>
    <button
      class="focus"
      disabled=""
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`focus blur via tab, tab, tab, click to focus button, click to blur button 5`] = `
<div
  id="__next"
>
  <div
    class="jsx-2517315232"
  >
    <div
      class="jsx-3659724228"
    >
      ·Slad·
    </div>
    <div
      class="jsx-442929638"
    >
      ·Basic Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          style="font-size:24px"
        >
          ·heading·
        </div>
        <div
          style="font-size:16px"
        >
          ·paragraph·
        </div>
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <div
      class="jsx-442929638"
    >
      ·Schema Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          class="jsx-3102962925"
        >
          ·heading·
        </div>
        <div
          class="jsx-1758464876"
        >
          ·paragraph·
        </div>
        <div
          class="jsx-44984361"
        >
          <div
            class="jsx-1758464876"
          >
            ·listitem·
          </div>
        </div>
        <img
          alt="Square placeholder image 80px"
          height="80"
          src="https://via.placeholder.com/80"
          width="80"
        />
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          ·{"anchor":[0,0,0],"focus":[0,0,0]}·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <button
      class="focus"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`render 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-2517315232"
  >
    <div
      class="jsx-3659724228"
    >
      ·Slad·
    </div>
    <div
      class="jsx-442929638"
    >
      ·Basic Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          style="font-size:24px"
        >
          ·heading·
        </div>
        <div
          style="font-size:16px"
        >
          ·paragraph·
        </div>
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <div
      class="jsx-442929638"
    >
      ·Schema Example·
    </div>
    <div
      autocorrect="off"
      contenteditable="true"
      data-gramm="true"
      role="textbox"
      spellcheck="false"
      style="background-color:#eee;margin-bottom:24px;outline:none;padding:8px;width:300px"
      tabindex="0"
    >
      <div>
        <div
          class="jsx-3102962925"
        >
          ·heading·
        </div>
        <div
          class="jsx-1758464876"
        >
          ·paragraph·
        </div>
        <div
          class="jsx-44984361"
        >
          <div
            class="jsx-1758464876"
          >
            ·listitem·
          </div>
        </div>
        <img
          alt="Square placeholder image 80px"
          height="80"
          src="https://via.placeholder.com/80"
          width="80"
        />
      </div>
    </div>
    <div>
      <div
        class="jsx-1141794554"
      >
        <pre>
          ·selection: ·
          <br />
          ·hasFocus: ·
          ·false·
        </pre>
      </div>
    </div>
    <button
      class="focus"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;
